docs/architecture.md

```diff
@@ -1,8 +1,94 @@
 # Project Helix Architecture
 
-![](./Helix%20Architecture/Helix%20Stack.png)
 
-## Preface
+
+## Architectural Evolution
+
+Helix has undergone some architectural evolution since the start of the project. The main phases are:
+
+1. Helix 1 – a bring-your-own (Adobe I/O) runtime framework-driven approach that would allow server-side customization
+2. [Helix 2](#helix-2) – a software as a service that relies on client-side customization
+3. [Helix 3](#helix-3) – a software as a service with superior availability and performance with the same feature set as Helix 2
+
+The current version of the architecture is Helix 3, prior versions stay documented as long as they are in active use.
+
+## Helix 3
+
+Helix 3 delivers the core feature set of Helix (high-performance websites with low code and zero-friction authoring) in a simplified architecture that optimizes stability and availability.
+
+![](./Helix%20Architecture/Helix%203%20Stack.png)
+
+### Preface
+
+Project Helix is based on serverless micro services architecture, where each service is managed in a separate GitHub repository. Each individual service is designed to be single-purpose and can typically be understood on its own within 15 minutes to half an hour. The connections between the different services however that make up the overall architecture require some explanation.
```

Comment:
> Each individual service is designed to be single-purpose

ideally: yes, in reality: no ;-)

the `admin service` is by design multi-purpose and the `pipeline service` isn't single-purpose anymore.